src/backend/distributed/transaction/backend_data.c

```diff
@@ -632,6 +633,7 @@ UnSetDistributedTransactionId(void)
 		MyBackendData->transactionId.transactionOriginator = false;
 		MyBackendData->transactionId.transactionNumber = 0;
 		MyBackendData->transactionId.timestamp = 0;
+		setDistributedTransactionId = false;
```

Comment:
Minor readability: We don't need to do this within spinlocks. Let's maybe do this right after we release the lock so that it becomes slightly more accurate in terms of its semantic.